zkevm-circuits/src/evm_circuit/execution/returndatasize.rs

```diff
@@ -0,0 +1,186 @@
+use crate::{
+    evm_circuit::{
+        execution::ExecutionGadget,
+        param::N_BYTES_MEMORY_ADDRESS,
+        step::ExecutionState,
+        util::{
+            common_gadget::SameContextGadget,
+            constraint_builder::{ConstraintBuilder, StepStateTransition, Transition::Delta},
+            from_bytes, CachedRegion, RandomLinearCombination,
+        },
+        witness::{Block, Call, ExecStep, Transaction},
+    },
+    table::CallContextFieldTag,
+    util::Expr,
+};
+use bus_mapping::evm::OpcodeId;
+use eth_types::{Field, ToLittleEndian};
+use halo2_proofs::plonk::Error;
+
+#[derive(Clone, Debug)]
+pub(crate) struct ReturnDataSizeGadget<F> {
+    same_context: SameContextGadget<F>,
+    return_data_size: RandomLinearCombination<F, N_BYTES_MEMORY_ADDRESS>,
```

Comment:
```suggestion
    return_data_size: RandomLinearCombination<F, N_BYTES_U64>,
```

Geth uses `uint64` for the length of return data:
https://github.com/ethereum/go-ethereum/blob/master/core/vm/instructions.go#L318

I think we should use `N_BYTES_U64`

Reply:
Make sense. Updated.

Reply:
I think that it should not be really necessary. As something bigger than U64 should imply that the memory expansion cost would be inassumible. 

And, so, we save some RLC saves for example.